### PR TITLE
Docs: Add steps for creating a new release

### PR DESCRIPTION
**Pull request by QuantumDancer**

### CODE REVIEW

File: media/website/content/development.md [hunk 1]
```diff
@@ -161,3 +161,45 @@ scripts/docs_pyauditor.sh
...
+- Create PR and wait for approval from other team member
+- Publish `auditor` crate first (you will need a [crates.io API token](https://crates.io/settings/tokens))
+  ```bash
+  cd auditor
+  cargo publish --dry-run
+  cargo publish
+  ```
+- Then run the publish workflow for all rust-based collectors/plugins (`cd` into corresponding dirs)
+  - `plugins/priority`
+  -  `collectors/slurm` (prepend `cargo` commands with `SQLX_OFFLINE=true`)
```
Comment (dirksammel) [inline media/website/content/development.md:191]: `SQLX_OFFLINE=true`, is this only needed for `collectors/slurm`?

**Review by dirksammel — COMMENTED** [review]
Should the Docker containers be mentioned?

File: media/website/content/development.md (hunk 1 above)
Comment (QuantumDancer) [reply media/website/content/development.md:191]: Yes

**Comment by QuantumDancer** [comment]
> Should the Docker containers be mentioned?

Added a sentence that they are automatically built and released once we push the tag.

**Review by dirksammel — APPROVED** [review]

**Pull request merged**